Add Support for UUID formatted Strings. Fixes #223

sbrunk opened the pull request:
I've taken a stab at UUID support (#223).

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

blast-hardcheese commented:
One other consideration, how many different variants of `UUID` are parsed by default? Even though this compiles, as that class represents a significant number of incompatible formats, I wonder how input submitted to servers, or values submitted to non-guardrail-generated servers is differentiated?

sbrunk commented:
- [X] Added to samples
- [X] Added `Show[UUID]`
- [X] Added missing `UUIDType` in JavaGenerator

As for the `UUID` variants and versions, AFAIK the Java version implements https://www.ietf.org/rfc/rfc4122.txt correctly so as long as the others do so as well this should be fine.

 [`UUID.fromString`](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/UUID.java#l191) seems actually quite permissive as long as the input size is ok and it can parse things as numbers.

blast-hardcheese commented:
Trying to push this over the line:
- Refactored how path extractors work a bit
- Broadened support for custom types
- Reduced the reliance on term matching at the edge
- Added a `Shower` test for UUID in Java
